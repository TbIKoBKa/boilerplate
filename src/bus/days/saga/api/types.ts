// Types
import { Days } from '../../types';

export type FetchDaysContract = () => Promise<Days>
