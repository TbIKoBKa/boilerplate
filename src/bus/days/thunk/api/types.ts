// Types
import { Days } from '../../types';

export type FetchDays = () => Promise<Days>
