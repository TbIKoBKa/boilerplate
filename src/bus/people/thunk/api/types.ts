// Types
import { People } from '../../types';

export type FetchPeople = () => Promise<People>
