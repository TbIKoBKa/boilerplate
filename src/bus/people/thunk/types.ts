// Fetch
export const FETCH_PEOPLE_ASYNC = 'FETCH_PEOPLE_ASYNC';
export type fetchPeopleActionAsync = {
    type: typeof FETCH_PEOPLE_ASYNC
}
export type FetchPeopleContact = () => fetchPeopleActionAsync
