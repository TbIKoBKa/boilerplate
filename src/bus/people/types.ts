// Core
export type Person = {
    name: string,
    height: string,
    mass: string,
    hair_color: string,
    skin_color: string,
    eye_color: string,
    birth_year: string,
    gender: string,
    homeworld: string,
    films: Array<string>,
    species: Array<string>,
    vehicles: Array<string>,
    starships: Array<string>,
    created: string,
    edited: string,
    url: string,
    [key: string]: any,
}

export type People = Array<Person>

// Set
export const SET_PEOPLE = 'SET_PEOPLE';
export type SetPeopleActionType = {
    type: typeof SET_PEOPLE;
    payload: People
}
export type SetPeopleContract = (payload: People) => SetPeopleActionType

export type PeopleActionTypes =
    | SetPeopleActionType
