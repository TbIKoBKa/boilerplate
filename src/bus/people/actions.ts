// Types
import * as types from './types';

export const setPeopleAction: types.SetPeopleContract = (payload) => ({
    type: types.SET_PEOPLE,
    payload,
});
