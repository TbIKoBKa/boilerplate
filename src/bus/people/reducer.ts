// Core
import { Reducer } from 'redux';

// Types
import * as types from './types';

const initialState: types.People = [];

export const peopleReducer: Reducer<types.People, types.PeopleActionTypes> = (state = initialState, action) => {
    switch (action.type) {
        case types.SET_PEOPLE:
            return action.payload;
        default:
            return state;
    }
};
