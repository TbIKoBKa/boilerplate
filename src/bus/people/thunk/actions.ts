import * as types from './types';

export const fetchPeopleActionAsync: types.FetchPeopleContact = () => ({
    type: types.FETCH_PEOPLE_ASYNC,
});
