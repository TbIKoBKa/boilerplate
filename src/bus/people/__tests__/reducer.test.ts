// Reducer
import { peopleReducer as people } from '../reducer';

// Actions
import { setPeopleAction } from '../actions';

// Types
import { People } from '../types';

const testPeople: People = [
    {
        name:       'Naruto',
        birth_year: '2002',
        created:    '2021',
        edited:     '2021',
        eye_color:  'blue',
        films:      [ 'Naruto', 'Boruto' ],
        gender:     'male',
        hair_color: 'yellow',
        height:     '176',
        homeworld:  'Ukraine',
        mass:       '73',
        skin_color: 'white',
        species:    [ 'species' ],
        starships:  [ 'Saske' ],
        url:        'kanoha.ua/users/narotuuzumaki',
        vehicles:   [ 'Hinata' ],
    },
];

describe('people reducer', () => {
    test('should handle set action', () => {
        expect(people(void 0, setPeopleAction(testPeople))).toMatchSnapshot();
    });
});
