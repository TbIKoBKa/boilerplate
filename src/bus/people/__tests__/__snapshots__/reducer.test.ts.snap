// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`people reducer should handle set action 1`] = `
Array [
  Object {
    "birth_year": "2002",
    "created": "2021",
    "edited": "2021",
    "eye_color": "blue",
    "films": Array [
      "Naruto",
      "Boruto",
    ],
    "gender": "male",
    "hair_color": "yellow",
    "height": "176",
    "homeworld": "Ukraine",
    "mass": "73",
    "name": "Naruto",
    "skin_color": "white",
    "species": Array [
      "species",
    ],
    "starships": Array [
      "Saske",
    ],
    "url": "kanoha.ua/users/narotuuzumaki",
    "vehicles": Array [
      "Hinata",
    ],
  },
]
`;
